waspc/data/Generator/templates/sdk/wasp/auth/password.ts

```diff
@@ -1,16 +1,36 @@
-import { Argon2id } from 'oslo/password'
+import { hash, verify, Version, type Options } from "@node-rs/argon2";
 
-const argon2id = new Argon2id()
+// The options are the same as the ones used in the oslo/password library
+const hashingOptions: Options = {
+  memoryCost: 19456,
+  timeCost: 2,
+  outputLen: 32,
+  parallelism: 1,
+  version: Version.V0x13,
+};
 
 // PRIVATE API
-export const hashPassword = async (password: string): Promise<string> => {
-  return argon2id.hash(password)
+export async function hashPassword(password: string): Promise<string> {
+  return hash(normalizePassword(password), hashingOptions);
```

Comment:
How do we know this is equivalent to what we had?

I know it follows the guide and you most likely tested it, but is it possible to unit test it somehow? I know our current setup isn't test-friendly, but maybe we can think of something (e.g., building an app and testing it there, like we did for operation type tests).

If it's too much work and nothing comes to mind, we can postpone until the testing effort which I hope is happening soon.

Reply:
Everything that we consider public API can be imported from userland and we could create a test app that unit tests all the APIs. Basically, write `.spec.ts` files for each piece of our public API and test it if possible. 

But this seems like a larger effort than these two functions and I'd do it when we do everything else.

Reply:
> How do we know this is equivalent to what we had?

I've tested it with an existing app (password hashed with oslo but checked by argon2 lib) and with a new app (password hashed and checkd by argon2 lib) and it worked. 

I've used what the oslo lib used internally, we are using the same code.

Reply:
> But this seems like a larger effort than these two functions and I'd do it when we do everything else.

Makes sense! Do you mind creating an issue?

Reply:
Here it is, please add any additional thoughts you might have on the topic: https://github.com/wasp-lang/wasp/issues/2221